cmd/common.go

```diff
@@ -138,14 +139,32 @@ func syncMain(filenames []string, dry bool, parallelism, delay int) error {
 }
 
 func kongVersion(config utils.KongClientConfig) (semver.Version, error) {
+
 	client, err := utils.GetKongClient(config)
 	if err != nil {
 		return semver.Version{}, err
 	}
 
 	root, err := client.Root(nil)
 	if err != nil {
-		return semver.Version{}, err
+		// try with workspace path
+		req, err := http.NewRequest("GET",
+			utils.CleanAddress(config.Address)+"/"+config.Workspace+"/kong",
+			nil)
+		if err != nil {
+			return semver.Version{}, err
+		}
+		var resp map[string]interface{}
+		_, err = client.Do(nil, req, &resp)
+		if err != nil {
+			return semver.Version{}, err
+		}
+
+		v, err := utils.CleanKongVersion(resp["version"].(string))
+		if err != nil {
+			return semver.Version{}, err
+		}
+		return semver.ParseTolerant(v)
```

Comment:
This piece of code is being repeated again at the bottom. Can you store the version from this call or the previous call on line 148 in a common variable and then use the code below? It would be better to avoid repetition here.

Reply:
Code optimized!